src/app/views/apps/modals/app-connect-modal-cloud.controller.coffee

```diff
@@ -0,0 +1,57 @@
+angular.module 'mnoEnterpriseAngular'
+  .controller('DashboardAppConnectCloudModalCtrl', ($scope, $uibModalInstance, app, MnoeAppInstances, toastr) ->
+
+    $scope.app = app
+    $scope.model = {}
+    $scope.hasLinked = false
+    $scope.hasChosenEntities = false
+    $scope.historicalData = false
+    $scope.isFormLoading = true
+    $scope.isSubmitting = false
+
+    MnoeAppInstances.getForm(app)
+      .then((response) ->
+        $scope.schema = response.schema
+        $scope.form = [ "*" ]
+        $scope.isFormLoading = false
+      )
+
+    $scope.getDate = ->
+      moment.locale('en')
+      moment().format("dddd, MMMM Do YYYY, h:mm:ss a")
+
+    $scope.historicalDataDisplay = ->
+      if document.getElementById('historical-data').checked
+        document.getElementById('historical-data-display-checked').style.display = 'block'
```

Comment:
(:pig:) Do not manipulate the DOM directly. This is not respecting MVC/MVVM separation of concerns. Use a `ng-model` in the template on the checkbox element.